app/src/libs/combat/tags.ts

```
@@ -47,19 +90,32 @@ export const absorb = (
               ? consequence.damage * (power / 100)
               : Math.min(power, consequence.damage);
           const convert = Math.ceil(absorbAmount * ratio);
+
           // Apply absorption to each pool
           pools.map((pool) => {
             switch (pool) {
               case "Health":
-                // Add to existing absorb value instead of overwriting
-                consequence.absorb_hp = (consequence.absorb_hp || 0) + convert / nPools;
+                // Calculate current HP absorption percentage
+                const currentAbsorbHp = consequence.absorb_hp || 0;
```

Biome (1.9.4)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Unsafe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)